streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -44,66 +49,120 @@ public static StoreFactory create(final String name) {
                 return new ValueFactory<K>() {
                     @Override
                     public <V> KeyValueFactory<K, V> withValues(final Serde<V> valueSerde) {
+
                         return new KeyValueFactory<K, V>() {
+                            public LoggedKeyValueFactory<K, V> logged(final String cleanupPolicy, final Map<String, String> config) {
+                                if (!InternalTopicManager.isValidCleanupPolicy(cleanupPolicy)) {
+                                    throw new IllegalArgumentException("cleanup policy: " + cleanupPolicy + " is not valid");
+                                }
+                                final Map<String, String> logConfig = new HashMap<>(config);
+                                logConfig.put("cleanup.policy", cleanupPolicy.toLowerCase(Locale.ROOT));
+                                return loggedKeyValueFactory(name, true, logConfig, keySerde, valueSerde);
+                            }
+
+                            public LoggedKeyValueFactory<K, V> notLogged() {
```

Comment:
For some coordinate with https://github.com/apache/kafka/pull/1752:

If we are going to add the `enableCaching` on the `persistentKeyValueFactory`, could we do the same here as well to be consistent? I.e. add the `enableLogging(final Map<String, String> config)` on the `persistentKeyValueFactory` as well as `InMemoryKeyValueFactory`, and get rid of the `LoggedKeyValueFactory`?